### PR TITLE
Travis CI: Upgrade to Python 3.7

### PR DESCRIPTION
**Pull request by cclauss**
Could it be a gcc difference?
* https://docs.travis-ci.com/user/reference/xenial gcc v5.4.0
* https://docs.travis-ci.com/user/reference/bionic gcc v7.4.0

### CODE REVIEW

**Comment by rgov** [comment]
Thanks for this one (and all the other pull requests)!

Debian "jessie" is on 3.4 and "stretch" is on 3.5. Ubuntu "Xenial" is on 3.5, "Bionic" and "Cosmic" are 3.6. macOS is on 2.7 and has been for a while, I think.

Could we instead do: 2.7, 3.5, 3.6, 3.7? This should cover most users.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+  - xenial
+  - bionic
+
 python:
   - "2.7"
   - "3.5"
   - "3.6"
   - "3.7"
 
+maxtrix:
```
Comment (rgov) [inline .travis.yml]: Probably `matrix`? Could you squash your changes? Thanks!

**Review by rgov — CHANGES_REQUESTED** [review]

**Comment by rgov** [comment]
What's happening with Bionic?

**Comment by cclauss** [comment]
Do you have a sense for why all the __bionic__ tests are failing?

**Comment by cclauss** [comment]
Could it be a gcc difference?

https://docs.travis-ci.com/user/reference/xenial gcc v5.4.0
https://docs.travis-ci.com/user/reference/bionic gcc v7.4.0

**Comment by rgov** [comment]
It's that `libbluetooth` is not available on Bionic. Is it a package you need to install, `libbluetooth-dev`?

**Comment by rgov** [comment]
I tried fixing this myself but ran into problems.

Personally, I am not a fan of Travis CI and I think that the container-based CI offered by CircleCI is superior. See also, https://github.com/pybluez/pybluez/issues/193 where Travis was also a problem.

**Comment by cclauss** [comment]
I propose that we check this one in with the bionic tests in __allow_failures__ mode to encourage experimentation to fix those tests.  I do not object to CircleCI but it would still be of interest to know why our Travis tests on bionic do not pass.

**Comment by rgov** [comment]
Thank you!

**Pull request merged**